com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/CollectionBuilderTest.java

```diff
@@ -914,4 +915,48 @@ public void testInstanceComponentCounterInSubCollectionFromFactory() throws Exce
             }
         }
     }
+
+    /**
+     * Test that the objects in the collection sorted according to its transform hierarhy
+     * Structure:
+     * - go "0"
+     *   - go "1"
+     *      - go "2"
+     *          - go "3"
+     * @throws Exception
+     */
+    @Test
+    public void testEmbeddedInstancesOrder() throws Exception {
+        StringBuilder src = new StringBuilder();
+        src.append("name: \"Example\"\n");
+        src.append("scale_along_z: 0\n");
+        src.append("embedded_instances {\n");
+        src.append("  id: \"3\"\n"); // Create the last object in hierarhy the first in the source file
```

Comment:
When a user creates objects in a collection editor, they are added one by one. It is possible that later the user may decide to move the very first object deep into the hierarchy.

Our current code in `gameobject.cpp` assumes that instances in `collection_desc` are sorted by transform hierarchy, allowing us to initialize them one by one and calculate the world_transform for each instance (and avoid calling an extra `UpdateTransform` every time a new collection is spawned).

In this case, object `3` is listed at the very beginning of the file but is deep in the hierarchy. The hierarchy is structured as follows:
- 0
  - 1
    - 2
      - **3**

However, in the file, the order is:
- **3**
- 0
- 1
- 2

When objects are spawned one by one without sorting, object `3` will be spawned and its world transform calculated at the very beginning, resulting in incorrect values. By sorting, all objects will be spawned layer by layer, ensuring the correct calculation of world transforms as each layer depends on the one above it.